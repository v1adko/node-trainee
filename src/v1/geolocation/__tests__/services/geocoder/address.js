import { geocoder } from '../../../';

jest.setTimeout(10000);

describe('Test the receiving coordinates from address', () => {
  it('should return coordinates', async () => {
    const coordinates = await geocoder.addressToCoordinates('kharkiv');
    expect(coordinates).toEqual([
      {
        address: 'Kharkiv, Kharkiv Oblast, Ukraine',
        coordinates: { lat: 49.9935, lon: 36.230383 }
      }
    ]);
  });

  it('should reject if no exist matched of address', async () => {
    const error = await geocoder.addressToCoordinates('');
    expect(error).toEqual({
      error: 'Error',
      message: 'Response status code is 400'
    });
  });
});
