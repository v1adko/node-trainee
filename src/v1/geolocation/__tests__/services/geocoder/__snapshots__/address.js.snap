// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the receiving coordinates from address should reject if no exist matched of address 1`] = `
Object {
  "error": "Error",
  "message": "Response status code is 400",
}
`;

exports[`Test the receiving coordinates from address should return coordinates 1`] = `
Array [
  Object {
    "address": "Kharkiv, Kharkiv Oblast, Ukraine",
    "coordinates": Object {
      "lat": 49.9935,
      "lon": 36.230383,
    },
  },
]
`;
